packages/grid/_modules_/grid/hooks/features/keyboard/useGridKeyboardNavigation.ts

```diff
@@ -0,0 +1,158 @@
+import { GridCellParams } from '../../../models/params/gridCellParams';
+import {
+  isArrowKeys,
+  isEnterKey,
+  isHomeOrEndKeys,
+  isPageKeys,
+  isSpaceKey,
+  isTabKey,
+} from '../../../utils/keyboardUtils';
+import { useGridApiMethod } from '../../root/useGridApiMethod';
+import { optionsSelector } from '../../utils/optionsSelector';
+import { visibleGridColumnsLengthSelector } from '../columns/gridColumnsSelector';
+import { useGridSelector } from '../core/useGridSelector';
+import { useGridState } from '../core/useGridState';
+import { gridPaginationSelector } from '../pagination/gridPaginationSelector';
+import { gridRowCountSelector } from '../rows/gridRowsSelector';
+import { useLogger } from '../../utils/useLogger';
+import { useGridApiEventHandler } from '../../root/useGridApiEventHandler';
+
+const getNextCellIndexes = (code: string, indexes: GridCellIndexCoordinates) => {
+  if (!isArrowKeys(code)) {
+    throw new Error('Material-UI: The first argument (code) should be an arrow key code.');
+  }
+
+  if (code === 'ArrowLeft') {
+    return { ...indexes, colIndex: indexes.colIndex - 1 };
+  }
+  if (code === 'ArrowRight') {
+    return { ...indexes, colIndex: indexes.colIndex + 1 };
+  }
+  if (code === 'ArrowUp') {
+    return { ...indexes, rowIndex: indexes.rowIndex - 1 };
+  }
+  // Last option code === 'ArrowDown'
+  return { ...indexes, rowIndex: indexes.rowIndex + 1 };
+};
+
+export const useGridKeyboardNavigation = (
+  gridRootRef: React.RefObject<HTMLDivElement>,
+  apiRef: GridApiRef,
+): void => {
+  const logger = useLogger('useGridKeyboardNavigation');
+  const options = useGridSelector(apiRef, optionsSelector);
+  const [, setGridState, forceUpdate] = useGridState(apiRef);
+  const paginationState = useGridSelector(apiRef, gridPaginationSelector);
+  const totalRowCount = useGridSelector(apiRef, gridRowCountSelector);
+  const colCount = useGridSelector(apiRef, visibleGridColumnsLengthSelector);
+  const containerSizes = useGridSelector(apiRef, gridContainerSizesSelector);
+
+  const mapKey = (event: React.KeyboardEvent) => {
+    if (isEnterKey(event.key)) {
+      return 'ArrowDown';
+    }
+    if (isTabKey(event.key)) {
+      return event.shiftKey ? 'ArrowLeft' : 'ArrowRight';
+    }
+    return event.key;
+  };
+
+  const navigateCells = React.useCallback(
+    (params: GridCellParams, event: React.KeyboardEvent) => {
+      const key = mapKey(event);
+      const isCtrlPressed = event.ctrlKey || event.metaKey || event.shiftKey;
+
+      const cellEl = params.element!;
+      cellEl.tabIndex = -1;
+
+      const currentColIndex = Number(cellEl.getAttribute('aria-colindex'));
+      const currentRowIndex = Number(cellEl.getAttribute('data-rowindex'));
+      const rowCount = options.pagination
+        ? paginationState.pageSize * (paginationState.page + 1)
+        : totalRowCount;
+
+      let nextCellIndexes: GridCellIndexCoordinates;
+      if (isArrowKeys(key)) {
```

Comment:
This looks like a difficult code to follow with the `if / else if`. Using `switch / case` will make it easier to read.

On the other hand - does it need to be a `if / else if`, can't it just be `if`s.

Reply:
Yes, it could be improved but I won't do it in this PR
It's quite big and not related to the edit mode, I just moved this function around to make the `useKeyboard` less cluttered.